VIP Scanner: filter_files() now accepts arrays of file types.

emrikol opened the pull request:
This should be a solution to GitHub issue #265.

I made this change so that when searching for XSS vulnerabilities, I wouldn't have to run through two `foreach ( $this->filter_files() )` blocks--one for PHP and one for HTML.

It should make scanning multiple types at once much easier.

Commit:
VIP Scanner: filter_files() now accepts arrays of file types for more efficient scanning.

I made this change so that when searching for XSS vulnerabilities, I wouldn't have to run through two foreach ( $this->filter_files() ) blocks.  One for php and one for html.
It should make scanning multiple types at once much easier.

nickdaugherty commented on vip-scanner/class-base-check.php:
```diff
@@ -55,11 +55,23 @@ protected function get_all_files( $files ) {
-				return array();
+		if ( $type ) {
+			if ( is_array( $type ) ) {
+				$file_types = array();
```
The naming of this variable is confusing, as we're not storing an array of file types, but rather files themselves.

Commit:
VIP Scanner: Renamed variable in filter_files() to make it more descriptive of its contents.

The variable $file_types had a name that could cause confusion.  Instead of storing file types, it stores files of multiple types (if $type is an array).
I renamed it to $files_of_multiple_types to be more in line with what the variable contains.
